packages/core/src/services/workspaces/usageOverview/getUsageOverview.ts

```diff
@@ -0,0 +1,92 @@
+import { desc, sql, eq, max } from 'drizzle-orm'
+import { database, Database } from '../../../client'
+import { buildDocumentLogsCountersQuery } from './documentLogsCountersQuery'
+import { buildEvaluationResultsCountersQuery } from './evaluationResultsCountersQuery'
+import { workspaceUsageInfoQuery } from './workspaceUsageInfoQuery'
+import { buildSubscriptionWithRenewalDatesQuery } from '../utils/calculateRenewalDate'
+
+/**
+ * NOTE:
+ * This query is pre-agrragating the data for the overview page.
+ * It has a subquery for documentLogs and evaluationResults.
+ * I'm aware that this can do full table scans.
+ * But without pre-aggregating the data the query results in duplicate rows.
+ * so the counts are wrong.
+ */
+export async function getUsageOverview(
+  {
+    pageSize,
+    page,
+    targetDate,
+  }: {
+    pageSize: number
+    page: number
+    targetDate?: Date
+  },
+  db: Database = database,
+) {
+  const renewalDates = buildSubscriptionWithRenewalDatesQuery(targetDate)
+  const documentLogsCountersQuery = buildDocumentLogsCountersQuery(renewalDates)
+  const evaluationResultsCountersQuery =
+    buildEvaluationResultsCountersQuery(renewalDates)
+  const query = db
```

Comment:
This query is not the more performance it could be. Or maybe yes. The problem it has is that I want aggregated counters for documentLogs and evaluationResults grouped by workspace->subscription calculated start period. 

I did 2 subqueries for these 2 aggregations and another subquery to do the calculated renewal date following the JS implementation. 

I would love to hear the best ideas. If any I would like to deploy this and see if runs with our current data volume.